dashboard/app/models/course.rb

```diff
@@ -24,6 +24,81 @@ def skip_name_format_validation
 
   include SerializedToFileValidation
 
+  def to_param
+    name
+  end
+
+  def self.file_path(name)
+    Rails.root.join("config/courses/#{name}.course")
+  end
+
+  def self.load_from_path(path)
+    serialization = File.read(path)
+    hash = JSON.parse(serialization)
+    course = Course.find_or_create_by!(name: hash['name'])
+    course.update_scripts(hash['script_names'])
+  rescue Exception => e
+    # print filename for better debugging
+    new_e = Exception.new("in course: #{path}: #{e.message}")
+    new_e.set_backtrace(e.backtrace)
+    raise new_e
+  end
+
+  # Updates courses.en.yml with our new localizeable strings
+  # @param name [string] - name of the course being updated
+  # @param course_strings[Hash{String => String}]
+  def self.update_strings(name, course_strings)
+    courses_yml = File.expand_path('config/locales/courses.en.yml')
+    i18n = File.exist?(courses_yml) ? YAML.load_file(courses_yml) : {}
+
+    i18n.deep_merge!({'en' => {'data' => {'course' => {'name' => {name => course_strings.to_h}}}}})
+    File.write(courses_yml, "# Autogenerated scripts locale file.\n" + i18n.to_yaml(line_width: -1))
+  end
+
+  def serialize
+    JSON.pretty_generate(
+      {
+        name: name,
+        script_names: course_scripts.map(&:script).map(&:name)
+      }
+    )
+  end
+
+  # This method updates both our localizeable strings related to this course, and
+  # the set of scripts that are in the course, then writes out our serialization
+  # @param scripts [Array<String>] - Updated list of names of scripts in this course
+  # @param course_strings[Hash{String => String}]
+  def update_and_persist(scripts, course_strings)
+    Course.update_strings(name, course_strings)
```

Comment:
Yeah, I wasn't doing it for thread safety. My intent was conceptual clarity. Strings for a course are all stored in a single file, so conceptually it made sense for me to have string persistence be a static method on the Course model rather than a method on the course instance.